iroh-api/src/api.rs

```diff
@@ -82,45 +93,66 @@ impl Api for Iroh {
         Ok(ClientP2p::new(p2p_client.clone()))
     }
 
-    async fn get<'b>(&self, ipfs_path: &IpfsPath, output: Option<&'b Path>) -> Result<()> {
-        let blocks = self.get_stream(ipfs_path, output);
-        tokio::pin!(blocks);
-        while let Some(block) = blocks.next().await {
-            let (path, out) = block?;
-            match out {
-                OutType::Dir => {
-                    tokio::fs::create_dir_all(path).await?;
+    fn get_stream(
+        &self,
+        ipfs_path: &IpfsPath,
+    ) -> LocalBoxStream<'_, Result<(RelativePathBuf, OutType)>> {
+        tracing::debug!("get {:?}", ipfs_path);
+        let resolver = iroh_resolver::resolver::Resolver::new(self.client.clone());
+        let results = resolver.resolve_recursive_with_paths(ipfs_path.clone());
+        let sub_path = ipfs_path.to_relative_string();
+        async_stream::try_stream! {
+            tokio::pin!(results);
+            while let Some(res) = results.next().await {
+                let (relative_ipfs_path, out) = res?;
+                let relative_path = RelativePathBuf::from_path(&relative_ipfs_path.to_relative_string())?;
+                // TODO this focusing in on sub-paths should really be handled in the resolver,
+                // because it'll be testable there, and it allows potential optimizations we
+                // can't do here
```

Comment:
can you write this TODO in this fashion:
`TODO(faassen)`

Also, I don't know if I agree with this. I would expect the path that comes out of the resolver be the path that it resolved the content from. I can see changing the output of the resolver from `PathBuf` to a full `IpfsPath`

Reply:
I attributed the todo and checked in a more elaborate argument. :)